pkg/gardenlet/controller/seed/seed_control.go

```diff
@@ -374,32 +374,38 @@ func (c *defaultControl) ReconcileSeed(obj *gardencorev1beta1.Seed, key string)
 
 func (c *defaultControl) updateSeedStatus(
 	ctx context.Context,
-	g gardencore.Interface,
+	cl client.Client,
 	seed *gardencorev1beta1.Seed,
 	k8sVersion string,
 	capacity, allocatable corev1.ResourceList,
 	updateConditions ...gardencorev1beta1.Condition,
 ) error {
-	if _, err := kutil.TryUpdateSeedStatus(ctx, g, retry.DefaultBackoff, seed.ObjectMeta,
-		func(seed *gardencorev1beta1.Seed) (*gardencorev1beta1.Seed, error) {
-			// remove "available condition"
-			for i, c := range seed.Status.Conditions {
-				if c.Type == "Available" {
-					seed.Status.Conditions = append(seed.Status.Conditions[:i], seed.Status.Conditions[i+1:]...)
-					break
-				}
-			}
+	seedCopy := seed.DeepCopy()
 
-			seed.Status.Conditions = gardencorev1beta1helper.MergeConditions(seed.Status.Conditions, updateConditions...)
-			seed.Status.ObservedGeneration = seed.Generation
-			seed.Status.Gardener = c.identity
-			seed.Status.KubernetesVersion = &k8sVersion
-			seed.Status.Capacity = capacity
-			seed.Status.Allocatable = allocatable
-			return seed, nil
-		},
-	); err != nil {
+	// remove "available condition"
+	for i, c := range seed.Status.Conditions {
+		if c.Type == "Available" {
```

Comment:
Do we still need that @rfranzke? Or does this even work with the `merge` strategy @BeckerMax?

Reply:
TLDR: The logic still works even with this merge strategy

I just tested it:
For testing I added a custom Type named `mytype`  to the conditions.  Then I changed the logic in line 387 to remove `mytype`.
The patchBytes then included the following
```
time="2021-02-05T20:59:35+01:00" level=info msg="{\"status\":{\"conditions\":[{\"$patch\":\"delete\",\"type\":\"mytype\"}]}}"
```
And the condition with type `mytype` indeed got removed from the conditions. So even with types.StrategicMergePatchType the explicit delete in the patch is respected.